modules/data.atmosphere/R/download.NOAA_GEFS_downscale.R

```diff
@@ -312,20 +333,20 @@ download.NOAA_GEFS_downscale <- function(outfolder, lat.in, lon.in, sitename, st
   #For each ensemble
   for (i in 1:21) { # i is the ensemble number
     #Generating a unique identifier string that characterizes a particular data set.
-    identifier = paste("NOAA_GEFS_downscale", sitename, i, format(start_date, "%Y-%m-%dT%H:%M"), 
-                       format(end_date, "%Y-%m-%dT%H:%M"), sep=".")
+    identifier = paste("NOAA_GEFS_downscale", site_id, i, format(start_date, "%Y-%m-%dT%H:%M"), 
+                       format(end_date, "%Y-%m-%dT%H:%M"), sep="_")
     
-    ensemble_folder = file.path(outfolder, identifier)
+    #ensemble_folder = file.path(outfolder, identifier)
     data = as.data.frame(joined %>% dplyr::select(NOAA.member, cf_var_names1) %>% 
                            dplyr::filter(NOAA.member == i) %>% 
                            dplyr::select(-NOAA.member))
     
-    #Each file will go in its own folder.
-    if (!dir.exists(ensemble_folder)) {
-      dir.create(ensemble_folder, recursive=TRUE, showWarnings = FALSE)
+  
+    if (!dir.exists(outfolder)) {
+      dir.create(outfolder, recursive=TRUE, showWarnings = FALSE)
     }
     
-    flname = file.path(ensemble_folder, paste(identifier, "nc", sep = "."))
+    flname = file.path(outfolder, paste(identifier, "nc", sep = "."))
```

Comment:
high level comment -- the code in this function looks identical to the code in the previous function, which is really bad practice. We should probably be handling the downloading and downscaling in separate functions (i.e. pull out the downscaling and have it be part of the general met_process workflow), as that's always been the design idea for met_process. As an interim, you might start by having download.NOAA_GEFS_downscale call download.NOAA_GEFS and then do the downscaling.

Reply:
Yes, we discussed this back when I created the downscale function and decided to do the whole other function for the interim. I don't think it's worth it to making another interim code since this one has been live for the last year. If I am going to go through the process of making the downscale call in the NOAA_GEFS function, I might as well make a generalized downscale function since I would have to deal with loading in the 21 nc files and downscaling and then re making the nc files. I did it this way last year because it's simpler to do the downscaling to the data before they are split by ensemble and made into nc files.

Reply:
OK, let's hold the reorganization off until the EFI RCN GEFS Harmonization discussion (Aug 21) since we'll inevitably need to reorganize code after that anyway.